hw/ip/flash_ctrl/doc/dv/index.md

```diff
@@ -87,35 +87,121 @@ All test sequences reside in `hw/ip/flash_ctrl/dv/env/seq_lib`.
+* flash_ctrl_intr_read
+  Task to program flash with interrupt mode.
+* flash_ctrl_intr_write
+  Task to read flash with interrupt mode.
+* send_rma_req
+  Task to initiate rma request. Once rma started, task polls `rma ack` until it completes.
 
 #### Functional coverage
 To ensure high quality constrained random stimulus, it is necessary to develop a functional coverage model.
 The following covergroups have been developed to prove that the test intent has been adequately met:
-* cg1:
-* cg2:
-
+`hw/ip/flash_ctrl/dv/env/flash_ctrl_env_cov.sv`
+* control_cg
+  Collects operation types, partition and cross coverage of both.
+* erase_susp_cg
+  Check if request of erase suspension occured.
+* msgfifo_level_cg
+  Covers all possible fifo status to generate interrupt for read / program.
+* rd_buff_evict_cg
+  Check sequence of operation to occur read eviction.
+* eviction_cg
+  Check whether eviction happens at all 4 caches with write / erase operation.
+  Also check each address belongs to randomly enabled scramble and ecc.
+* error_cg
+  Check errors degined in error code registers.
 ### Self-checking strategy
 #### Scoreboard
-The `flash_ctrl_scoreboard` is primarily used for end to end checking.
-It creates the following analysis ports to retrieve the data monitored by corresponding interface agents:
-* analysis port1:
-* analysis port2:
-<!-- explain inputs monitored, flow of data and outputs checked -->
+The `flash_ctrl_scoreboard` is primarily used for csr transaction integrity.
+Test bench also maintain reference momory model (associative array) per partition.
+The referrence model is updated each operation and used for expected partition values
+at the end of test check.
+Given large memory size (mega bytes), maintaining associative array per operation until the end of the test
+causes hugh simulation overhead. Also, this model is not suitible for read only test, same address write test
+, descramble tests and error injection test for ecc. To address such issues,
+`on-the-fly` method is also used on top of legacy test component.
+In `on-the-fly` test mode, test is not rely on reference memory mode. Instead, it creates reference data
+per each operation. For the program(write) operation, rtl output is collected from flash phy interface and
+compared with pre calucated write data (per each operation). For the read operation,
+expected data is written via memory backdoor interface and read back and compared.
+The `flash_ctrl_otf_scoreboard` is used for `on-the-fly` mode flash transaction integrity check while
+`flash_ctrl_scoreboard` is still used for csr transaction integrity check.
+The `last mile scoreboard` is added to compensate `on-the-fly` model. For the write transaction,
+`on-the-fly` model collects rtl data at the bounday of the controller and flash model.
+There are uncovered logic stil within the model before data reaches actual device. So we add
```

Comment:
```suggestion
There is still uncovered logic within the model before data reaches the actual device. So we add
```
I think this and the following sentence should go ahead of line 143, since it provides a justification for the "last mile scoreboard". Some other rearrangement may be needed once you do that.

Reply:
moved to line 144